File: fidanys-server/src/main/java/com/fidanlik/fidanysserver/expense/model/Expense.java
package com.fidanlik.fidanysserver.expense.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Document(collection = "expenses")
public class Expense {
    @Id
    private String id;
    private String tenantId;
    private String userId;

    private String description;
    private BigDecimal amount;
    private LocalDate expenseDate;

    // YENİ EKLENEN: Kategori ID'sini tutmak için string alan
    private String categoryId;

    @DBRef
    private ExpenseCategory category;

    private String paymentId; // Giderin hangi ödeme ile yapıldığını belirtir

    private String productionBatchId; // Bu giderin ilişkili olduğu üretim partisinin ID'si
}

File: client/src/types/expense.ts
// client/src/types/expense.ts

// Server tarafındaki ExpenseCategory modeline karşılık gelir
export interface ExpenseCategory {
  id: string;
  name: string;
  description?: string;
  tenantId: string;
}

// Server tarafındaki Expense modeline karşılık gelir
export interface Expense {
  id: string;
  tenantId: string;
  userId: string;
  description: string;
  amount: number; // BigDecimal yerine number
  expenseDate: string; // LocalDate yerine ISO string
  categoryId: string; // Kategori ID'si
  productionBatchId?: string; // Giderin ilişkili olduğu üretim partisinin ID'si (opsiyonel)
}

// Server tarafındaki PaymentMethod modeline karşılık gelir (varsayım)
export interface PaymentMethod {
  id: string;
  name: string;
  // Diğer ödeme yöntemleri özellikleri
}